### ISSUE TITLE
QNN backend static input length and kv-caching question

### ISSUE THREAD

**Issue by leigao97** [pytorch/executorch#2922]
Based on the [comment](https://github.com/pytorch/executorch/blob/3e256ffa676d2be3848decd1a8234cebcc8c9634/examples/models/llama2/export_llama_lib.py#L619) from the llama2 example:

"Qualcomm backend currently only supports static shape and use_kv_cache=True is the only way to support it at the moment"

Can you please elaborate more on this restriction and explain what the challenges are to support varied input length in QNN? Is there any way to work around this?

**Comment by cccclai** [comment]
Hi, thank you for checking out, qnn backend only support static shape at the moment. It's still under discussion to support dynamic shape. Also, using dynamic shape usually means less performant. What is your use case?

**Comment by leigao97** [comment]
Hi @cccclai, thank you for getting me back. I am trying to calculate the loss value given the input sequence. Do you think I can disable kv-cache and implement the loss function while still forcing the static shape of all tensors?

**Comment by cccclai** [comment]
oh if you are able to use static shape in local change, it's totally fine to disable the kv-cache.

**Comment by leigao97** [comment]
Thank you for the help.
